File: app/src/main/java/com/example/youtubeapiintegration/TopGravityDrawable.java
package com.example.youtubeapiintegration;

import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.drawable.BitmapDrawable;

public class TopGravityDrawable extends BitmapDrawable {

    public TopGravityDrawable(Resources res, Bitmap bitmap) {
        super(res, bitmap);
    }

    @Override
    public void draw(Canvas canvas) {
        int halfCanvas = canvas.getHeight() / 2;
        int halfDrawable = getIntrinsicHeight() / 2;
        canvas.save();
        canvas.translate(0, -halfCanvas + halfDrawable);
        super.draw(canvas);
        canvas.restore();
    }
}

File: app/src/main/java/com/example/youtubeapiintegration/TopGravityDrawable.kt
package com.example.youtubeapiintegration

import android.content.res.Resources
import android.graphics.Bitmap
import android.graphics.Canvas
import android.graphics.drawable.BitmapDrawable

class TopGravityDrawable(res: Resources, bitmap: Bitmap) : BitmapDrawable(res, bitmap) {

    override fun draw(canvas: Canvas) {
        val halfCanvas = bounds.height() / 2
        val halfDrawable = intrinsicHeight / 2
        canvas.save()
        canvas.translate(0f, (-halfCanvas + halfDrawable).toFloat())
        super.draw(canvas)
        canvas.restore()
    }
}
